Mathlib/NumberTheory/ArithmeticFunction.lean

```diff
@@ -591,6 +595,33 @@ theorem pdiv_zeta [DivisionSemiring R] (f : ArithmeticFunction R) :
 
 end Pdiv
 
+section ProdToFinsetFactors
+
+def prodToFinsetFactors [CommMonoidWithZero R] (f : ℕ → R) : ArithmeticFunction R :=
+  ⟨fun d => if d = 0 then 0 else ∏ p in d.factors.toFinset, f p, if_pos rfl⟩
+
+open Std.ExtendedBinder
+
+scoped syntax (name := bigproddvd) "∏ᵖ " extBinder " ∣ " term ", " term:67 : term
+scoped macro_rules (kind := bigproddvd)
+  | `(∏ᵖ $x:ident ∣ $n, $r) => `(prodToFinsetFactors (fun $x ↦ $r) $n)
+
+scoped syntax (name := bigproddvdarith) "∏ᵖ " term:67 : term
+scoped macro_rules (kind := bigproddvdarith)
+  | `(∏ᵖ $f) => `(prodToFinsetFactors $f)
+
+@[simp]
+theorem prodToFinsetFactors_apply [CommMonoidWithZero R] {f: ℕ → R} {n : ℕ} [hn : NeZero n] :
```

Comment:
This looks like a mis-use of `NeZero` to me

Reply:
I chose to do this based on a [suggestion](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/finite.20product.20of.20infinite.20sums/near/379584701) by @kbuzzard, and it does come in handy for `prodToFinsetFactors` in that (a) it works with `simp_rw` and (b) it automatically deduces `NeZero (x*y)`